tests/cupy_tests/math_tests/test_sumprod.py

```diff
@@ -715,6 +716,45 @@ def test_cumprod_numpy_array(self, dtype):
             return cupy.cumprod(a_numpy)
 
 
+@testing.parameterize(*testing.product({
+    'shape': [(20,), (7, 6), (3, 4, 5)],
+    'dtype': ['float32', 'float64', 'complex64', 'complex128'],
```

Comment:
I think we should test for all dtypes, including integers. Obviously integers wouldn't have NaN, but NumPy covers it...

Reply:
OK, I will add tests for integer dtypes (but without NaN..)